staging/src/k8s.io/client-go/discovery/aggregated_discovery.go

```diff
@@ -55,8 +56,12 @@ func convertAPIGroup(g apidiscovery.APIGroupDiscovery) (*metav1.APIGroup, map[sc
 	gvResources := map[schema.GroupVersion]*metav1.APIResourceList{}
 	group.Name = g.ObjectMeta.Name
 	for i, v := range g.Versions {
-		version := metav1.GroupVersionForDiscovery{}
 		gv := schema.GroupVersion{Group: g.Name, Version: v.Version}
+		if v.Freshness == apidiscovery.DiscoveryFreshnessStale {
+			klog.V(5).Infof("stale group/version omitted from discovery: %v", gv)
+			continue
```

Comment:
In the legacy scenario, we return something in the error `&ErrorGroupDiscoveryFailed{Groups: failedGroups}` https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/discovery/discovery_client.go#L428 for any gvs that were unreachable. Can we propagate this information to `ServerGroupsAndResources()` in the aggregated discovery as well?

Reply:
Addressed. Lots of plumbing complexity. Let me know what you think.

Reply:
Does it make sense to return the resources even if the gv is stale? I noticed that's what we do currently https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/discovery/discovery_client.go#L531, though not really sure about the use case for it.

Reply:
This question still stands but I don't know what the answer should be. Mostly pertains to aggregated apiservers being unavailable since local apiservers almost always will not encounter this. cc @deads2k

Reply:
Ah, the resource list will always be nil in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/discovery/discovery_client.go#L529, thanks for confirming! Yeah I agree the comment is wrong, we shouldn't ever hit that case.

Reply:
Not having resources for stale discovery information seems valid since otherwise we will not converge with kube-apiservers that have restarted.  If we eventually have a persisted set of resources so stale resources would be consistent among all kube-apiservers, I'd be in favor of doing so.